### PR TITLE
Add new formatter and run it

### PR DESCRIPTION
**Pull request by cltnschlosser**
Probably touched every file, but now we will have a formatter to run.

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/blockout/views/ScrollingView.java [hunk 1]
```diff
@@ -57,6 +46,17 @@ protected ScrollingContainer createScrollingContainer()
         return new ScrollingContainer(this);
     }
 
+    /**
+     * Load from xml.
+     *
+     * @param params xml parameters.
+     */
+    public ScrollingView(PaneParams params)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/blockout/views/ScrollingView.java]: MINOR Move this constructor to comply with Java Code Conventions. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1213)

File: src/main/java/com/minecolonies/colony/permissions/Permissions.java [hunk 2]
```diff
@@ -22,118 +22,22 @@
...
-
-    private static final String TAG_OWNERS = "owners";
-    private static final String TAG_ID = "id";
-    private static final String TAG_RANK = "rank";
-    private static final String TAG_PERMISSIONS = "permissions";
-    private static final String TAG_FLAGS = "flags";
-    private static Map<Rank, RankPair> promotionRanks = new EnumMap<>(Rank.class);
-    private Map<UUID, Player> players = new HashMap<>();
+    private Map<UUID, Player>  players     = new HashMap<>();
     private Map<Rank, Integer> permissions = new EnumMap<>(Rank.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/permissions/Permissions.java]: MAJOR Rename field "permissions" [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1700)

File: src/main/java/com/minecolonies/colony/permissions/Permissions.java [hunk 3]
```diff
@@ -255,54 +231,43 @@ public void savePermissions(NBTTagCompound compound)
...
-     * @return          Rank of te player
-     */
-    public Rank getRank(EntityPlayer player)
-    {
-        return getRank(player.getGameProfile().getId());
-    }
-    /**
+                .filter(player -> ranks.contains(player.rank))
+                .collect(Collectors.toSet());
+    }    /**
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/permissions/Permissions.java]: INFO Move this trailing comment on the previous empty line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 10 issues
* MAJOR 3 major
* MINOR 2 minor
* INFO 5 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [OpenInventoryMessage.java#L95](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/network/messages/OpenInventoryMessage.java#L95): Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1151)
1. MAJOR [OpenInventoryMessage.java#L103](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/network/messages/OpenInventoryMessage.java#L103): Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1151)
1. MINOR [SchematicWrapper.java#L117](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/util/SchematicWrapper.java#L117): Move this constructor to comply with Java Code Conventions. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1213)
1. INFO [InventoryCitizen.java#L338](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/inventory/InventoryCitizen.java#L338): Move this trailing comment on the previous empty line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. INFO [InventoryCitizen.java#L380](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/inventory/InventoryCitizen.java#L380): Move this trailing comment on the previous empty line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. INFO [InventoryCitizen.java#L402](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/inventory/InventoryCitizen.java#L402): Move this trailing comment on the previous empty line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. INFO [TileEntityColonyBuilding.java#L74](https://github.com/Minecolonies/minecolonies/blob/538e10e77d98308137659d21605dc59e1f003c5a/src/main/java/com/minecolonies/tileentities/TileEntityColonyBuilding.java#L74): Move this trailing comment on the previous empty line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by Kostronor** [comment]
Why run it on all files @cltnschlosser 
There were some plans to go through the files one by one and fix code style as per the formatter. Would make a lot less merge conflicts imo

**Comment by cltnschlosser** [comment]
@Kostronor We don't need to merge this, but this is a pretty good sample of what this formatter would do. Also plan on running it on the whole thing at some point, probably after a release when everyone has just finished what they were working on. No point in making a formatter if we go one by one.

**Comment by Kostronor** [comment]
I'll run the formater on our way to 1.10.2

**Pull request closed without merging**